backend/app/api/v1/endpoints/company.py

```diff
@@ -26,3 +26,16 @@ async def get_all_companies(db_session=Depends(get_db)):
     """
     
     return CompanyRepository(db_session).get_all_companies()
+
+@router.get("/{id}", status_code=status.HTTP_200_OK, response_model=Company)
+async def get_company_by_id(id: str, db_session=Depends(get_db)):
```

Comment:
heh depends. Maybe ill get the records, maybe i wont. depends

Reply:
lol, depends if you have the db session!